serde_codegen_internals/src/attr.rs

```diff
@@ -797,3 +894,20 @@ fn parse_lit_into_where(cx: &Ctxt,
 
     syn::parse_where_clause(&where_string).map(|wh| wh.predicates).map_err(|err| cx.error(err))
 }
+
+fn parse_lit_into_ty(cx: &Ctxt,
+                     attr_name: &str,
+                     meta_item_name: &str,
+                     lit: &syn::Lit)
+                     -> Result<syn::Ty, ()> {
+    let string = try!(get_string_from_lit(cx, attr_name, meta_item_name, lit));
+    if string.is_empty() {
+        return Ok(syn::Ty::Tup(Vec::new()));
```

Comment:
Please remove this special case. `from = ""` does not make sense and should be an error just like any other type that does not parse.